beacon_node/http_api/src/block_id.rs

```diff
@@ -211,6 +211,51 @@ impl BlockId {
             }
         }
     }
+
+    /// Return the `BlobsSidecar` identified by `self`.
+    pub async fn blobs_sidecar<T: BeaconChainTypes>(
+        &self,
+        chain: &BeaconChain<T>,
+    ) -> Result<(Arc<BlobsSidecar<T::EthSpec>>), warp::Rejection> {
+        let root = match &self.0 {
+            CoreBlockId::Head => {
+                let (cached_head, _execution_status) = chain
+                    .canonical_head
+                    .head_and_execution_status()
+                    .map_err(warp_utils::reject::beacon_chain_error)?;
+                cached_head.head_block_root()
+            }
+            CoreBlockId::Slot(slot) => {
+                let maybe_block_root = chain
+                    .block_root_at_slot(*slot, WhenSlotSkipped::None)
+                    .ok()
+                    .flatten();
+                match maybe_block_root {
+                    Some(block_root) => block_root,
+                    None => {
+                        return Err(warp_utils::reject::custom_not_found(format!(
+                            "Block root for slot {} not found",
+                            slot
+                        )))
+                    }
+                }
+            }
+            _ => self.root(chain)?.0,
```

Comment:
I think it'd be ok to use `self.root(chain)?.0` for all `CoreBlockId` variants, wondering what the thinking here was for overriding `Head` and `Slot`? 

`let root = self.root(chain)?.0;`

Reply:
ah great point, that simplifies things a lot!

Initially I wasn't aware of the `root` function, and was trying to somehow get the `root` using just `chain` and `CoreBlockId`..  and when I implemented the `_` arm, I the didn't realise it would work for the other variants too.

Applying your suggestion, thanks!